subsys/caf/modules/leds.c

```diff
@@ -175,6 +183,19 @@ static int leds_init(void)
 
 	BUILD_ASSERT(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT) > 0, "No LEDs defined");
 
+	/* Avoid referring to the same Zephyr LED device with multiple CAF LED instances. */
+	if (IS_ENABLED(CONFIG_ASSERT) && IS_ENABLED(CONFIG_CAF_LEDS_GPIO)) {
```

Comment:
I don't like this part. What it is for?

Reply:
Zephyr boards by default do not define a label for `gpio-leds` compatible node. The node name is set to `leds` (default naming convention in Zephyr) and the name is used to get driver's binding. I wanted to make CAF LEDs compatible with default naming convention used in Zephyr (in similar way as it's compatible for `pwm_leds`). The user can still define custom nodes with user-defined labels.

Reply:
Unfortunately LED drivers using GPIO and PWM use various convention when selecting default driver name (the one used in `led_pwm` was not accepted in `led_gpio` - changes were requested).

Reply:
Maybe we could demand from CAF user to provide a label for every LED and then always use user-defined label? In that case the default labels would not be used by both GPIO and PWM LEDs and behavior will be more consistent.

Reply:
discussed offline